src/game/services/Rules/state-machine/playerWateringCropState.ts

```diff
@@ -90,14 +88,18 @@ export const playerWateringCropState: RulesMachineConfig['states'] = {
           }
 
           case GameEvent.OPERATION_ABORTED: {
-            selectedWaterCardInHandIdx = defaultSelectedWaterCardInHandIdx
             break
           }
 
           default:
         }
 
-        enqueue.assign({ game, selectedWaterCardInHandIdx })
+        enqueue.assign({
+          game: {
+            ...game,
+            selectedWaterCardInHandIdx: defaultSelectedWaterCardInHandIdx,
```

Comment:
It seems like `selectedWaterCardInHandIdx` will always be set to `defaultSelectedWaterCardInHandIdx` here. This seems like a potential regression. If that's the case, can we restore the previous logic of only setting `selectedWaterCardInHandIdx` to `defaultSelectedWaterCardInHandIdx` if the operation was aborted?